hoptimator-k8s/src/main/java/com/linkedin/hoptimator/k8s/K8sContext.java

```diff
@@ -56,9 +63,18 @@ public SharedInformerFactory informerFactory() {
     return informerFactory;
   }
 
+  public String watchNamespace() {
+    return watchNamespace;
+  }
+
+  public void watchNamespace(String watchNamespace) {
+    this.watchNamespace = watchNamespace;
```

Comment:
I think it's best if `currentContext()` is a global variable / singleton, without any setters. It's unpredictable otherwise. e.g. you could invoke `watchNamespace("foo")` and have no idea if your change is before or after the reconcilers start listening.